great_expectations/rule_based_profiler/domain_builder/categorical_column_domain_builder.py

```diff
@@ -0,0 +1,247 @@
+            max_proportion_unique: proportion of unique values for a
+                custom cardinality limit to use when filtering columns.
+            exclude_columns: If provided, these columns are pre-filtered and
+                excluded from consideration, cardinality is not computed.
+        """
+
+        super().__init__(
+            data_context=data_context,
+            batch_request=batch_request,
+        )
+
+        self._cardinality_checker = CardinalityChecker(
+            limit_mode=limit_mode,
+            max_unique_values=max_unique_values,
+            max_proportion_unique=max_proportion_unique,
+        )
+
+        self._exclude_columns = exclude_columns
+
+    @property
+    def domain_type(self) -> Union[str, MetricDomainTypes]:
+        return MetricDomainTypes.COLUMN
+
+    def _get_domains(
+        self,
+        variables: Optional[ParameterContainer] = None,
+    ) -> List[Domain]:
+        """Return domains matching the selected limit_mode.
+
+        Args:
+            variables: Optional variables to substitute when evaluating.
+
+        Returns:
+            List of domains that match the desired cardinality.
+        """
+
+        batch_ids: List[str] = self._get_batch_ids(variables=variables)
+        validator: Validator = self.get_validator(variables=variables)
+
+        # Here we use a single get_metric call to get column names to build the
+        # rest of the metrics.
+        table_column_names: List[str] = self._get_table_column_names_from_first_batch(
+            validator=validator,
+            batch_ids=batch_ids,
+        )
+
+        metrics_for_cardinality_check: List[
+            Dict[str, List[MetricConfiguration]]
+        ] = self._generate_metric_configurations_to_check_cardinality(
+            batch_ids=batch_ids, column_names=table_column_names
+        )
+
+        candidate_column_names: List[str] = self._columns_meeting_cardinality_limit(
+            validator=validator,
+            table_column_names=table_column_names,
+            metrics_for_cardinality_check=metrics_for_cardinality_check,
+        )
+
+        column_domains: List[Domain] = build_domains_from_column_names(
+            candidate_column_names
+        )
+
+        return column_domains
+
+    def _get_table_column_names_from_first_batch(
+        self,
+        validator: Validator,
+        batch_ids: List[str],
+    ):
+        """Retrieve table column names from the first loaded batch.
+
+        Args:
+            validator: Validator to use in retrieving columns.
+            batch_ids: Batches used in this profiler rule.
+
+        Returns:
+            List of column names from the first loaded batch.
+        """
+
+        if len(batch_ids) == 0:
+            raise ProfilerConfigurationError(
+                "No batch_ids were available, please check your configuration."
+            )
+
+        first_batch_id: str = batch_ids[0]
```

Comment:
@anthonyburdi Why not just used `get_batch_id()` for this?  Efficiency?  Perhaps -- but please note that the first `Batch` should probably be the `active_batch` -- and the index for it is: -1 (negative 1) -- thanks!

Reply:
Changed in a local commit, will push after resolving other comments.